src/MultiReport/Keyboard.cpp

```diff
@@ -139,14 +139,36 @@ int Keyboard_::sendReportUnchecked(void) {
 
 
 int Keyboard_::sendReport(void) {
+  // ChromeOS 51-60 (at least) bug: if a modifier and a normal keycode are added in the
+  // same report, in some cases the shift is not applied (e.g. `shift`+`[` doesn't yield
+  // `{`). To compensate for this, check to see if the modifier byte has changed. If so,
+  // copy the modifier byte to the previous key report, and resend it before proceeding.
+  if (lastKeyReport.modifiers != keyReport.modifiers) {
+    uint8_t last_mods = lastKeyReport.modifiers;
+    lastKeyReport.modifiers = keyReport.modifiers;
+    int returnCode = HID().SendReport(HID_REPORTID_NKRO_KEYBOARD, &lastKeyReport, sizeof(lastKeyReport));
+    if (returnCode < 0)
+      lastKeyReport.modifiers = last_mods;
+    // For Windows Remote Desktop, the problem is even worse. Even if the modifier is sent
+    // in a separate report, if one or more other keycodes are added in a subsequent
+    // report that comes too soon (probably before the next "frame" is sent to the remote
+    // host), it seems that the two reports get combined, and we once again see the
+    // problem. So, if both a modifier keycode and a non-modified keycode have changed in
+    // one report, we add a delay between the modifier report (sent above) and the other
+    // report (sent below). 10ms seems to work well.
+    if (memcmp(lastKeyReport.keys, keyReport.keys, sizeof(keyReport.keys)))
+      delay(10);
```

Comment:
Putting an explicit delay in here is, at first blush, a big, big red flag to me.  If we do that, it means that during those 10ms, the keyboard is effectively dead in the water. We've worked really hard to eliminate delays hardcoded anywhere near the core after bootup time. (Right now, I see one in Macros and one in Unicode hex input.)
Can you break the Windows RDP workaround out into a separate pull request so we can merge the ChromeOS fix?

Reply:
Sure thing. I'm not exactly fond of that delay, either, and it's related, but not exactly the same problem. I'll separate that into a different PR tonight.